hip_data_tools/google/adwords.py

```diff
@@ -262,3 +322,200 @@ def _verify_accepted_columns(self, conversion: dict) -> None:
                 raise ValueError(
                     f"The column {col} present in the DataFrame is not in the allowed column list "
                     f"{self.valid_fields}")
+
+
+def _get_page_as_dict(page) -> List[dict]:
+    if 'entries' in page:
+        return page['entries']
+    else:
+        log.warning('No entries were found.')
+        return []
+
+
+class AdWordsCampaignUtil(AdWordsUtil):
+    """
+    Adwords Utility to handle Campaign details
+    Args:
+        conn (GoogleAdWordsConnectionManager): Connection manager to handle the creation of
+        adwords client
+    """
+
+    def __init__(self, conn: GoogleAdWordsConnectionManager, page_size: int = 500):
+        super().__init__(conn, service='CampaignService', version='v201809')
+        self.page_size = page_size
+
+    def set_query_to_fetch_all(self) -> None:
+        """
+        Get all campaigns associated with and adwords account query set on utility
+        Returns: None
+        """
+        query = (ServiceQueryBuilder()
+                 .Select('Id', 'Name', 'Status')
+                 # .Where('Status').EqualTo('ENABLED')
+                 .OrderBy('Id')
+                 .Limit(0, self.page_size)
+                 .Build())
+        self.set_query(query)
+
+
+class AdWordsAdGroupUtil(AdWordsUtil):
+    """
+    Adwords Utility to handle AdGroup details
+    Args:
+        conn (GoogleAdWordsConnectionManager): Connection manager to handle the creation of
+        adwords client
+    """
+
+    def __init__(self, conn: GoogleAdWordsConnectionManager, page_size: int = 500):
+        super().__init__(conn, service='AdGroupService', version='v201809')
+        self.page_size = page_size
+        self._all_query = None
+
+    def set_query_to_fetch_by_campaign(self, campaign_id: str) -> None:
+        """
+        Get Ad groups for the given Campaign query set on utility
+        Args:
+            campaign_id (str): the adwords campaign to query
+        Returns: None
+        """
+        query = (ServiceQueryBuilder()
+                 .Select('Id', 'CampaignId', 'CampaignName', 'Status')
+                 .Where('CampaignId').EqualTo(campaign_id)
+                 .OrderBy('Id')
+                 .Limit(0, self.page_size)
+                 .Build())
+        self.set_query(query)
+
+    def set_query_to_fetch_all(self) -> None:
+        """
+        Get all Ad groups in the account query set on utility
+        Returns: None
+        """
+        query = (ServiceQueryBuilder()
+                 .Select('Id', 'CampaignId', 'CampaignName', 'Status', 'Settings', 'Labels',
```

Comment:
are these all the possible columns? Should this be a settable variable?

Reply:
yes, 
this is a convenient shortcut method done here as an example, you can customise the query and write your own, and define any column you like, using the set_query method.